internal/sidekick/parser/httprule/http_rule_parser.go

```diff
@@ -91,6 +91,11 @@ func ParseSegments(pathTemplate string) (*api.PathTemplate, error) {
 //
 // This is different from `ParseSegments` because it does not expect a leading `/`.
 func ParseResourcePattern(pathTemplate string) (*api.PathTemplate, error) {
+	// TODO(https://github.com/googleapis/librarian/issues/3258): ParseResourcePattern
+	// should support parsing generic resources more robustly than just checking for a literal `*`.
+	if pathTemplate == api.SingleSegmentWildcard {
+		return api.NewPathTemplate().WithLiteral(api.SingleSegmentWildcard), nil
```

Comment:
This will only match a string that is exactly `"*"`, right? I don't think a literal makes sense.

I don't think there's a good representation only using path segments (which are all named variables).

You could try something like a:

```
{
  MatchAny: bool,
  Path: []PathSegment
}
```

... but then you have to make sure everything downstream of this knows what `MatchAny` means.